File: templates/python01/file_util.py

#######################################################
# 'r'：读
# 'w'：写
# 'a'：追加
# 'r+' == r+w（可读可写，文件若不存在就报错(IOError)）
# 'w+' == w+r（可读可写，文件若不存在就创建）
# 'a+' == a+r（可追加可写，文件若不存在就创建）
# 对应的，如果是二进制文件，就都加一个b就好啦：
# 'rb'　　'wb'　　'ab'　　'rb+'　　'wb+'　　'ab+'
#######################################################


# 逐行处理文件， fun 是传入的处理函数
# 比较省内存，处理特大文件用
def processFile(filename, fun=None, stripLine=False, encoding='utf-8'):
    file = open(filename, 'r', encoding=encoding) 
    while 1:
        line = file.readline() # line 是带换行符的
        if(stripLine):
            line = line.strip("\n")
        if not line:
            break
        if(fun != None):
            fun(line)
    file.close()

# 另一种逐行读入的处理方法
def processLine(filename, encoding='utf-8'):
    file = open(filename, 'r', encoding=encoding) 
    for line in file:
        pass # do something
    file.close()

# 按行读取并返回处理后的对象列表
# fun 是对行进行处理的函数
def readAndProcess(filename, fun, skipNone=True, encoding='utf-8'):
    result = []
    file = open(filename, 'r', encoding=encoding)
    lines = file.readlines()
    for line in lines:
        line = line.strip('\n')
        item = fun(line)
        if skipNone and item == None:
            pass
        else:
            result.append(item)
    return result


# 返回文件所有行的列表，去掉换行符
def readLines(filename, encoding='utf-8'):
    result = []
    file = open(filename, 'r', encoding=encoding) 
    lines = file.readlines()
    for line in lines:
        result.append(line.strip('\n'))

    file.close()
    return result

# 返回文件内容的字符串
def readFile(filename, encoding='utf-8'):
    file = open(filename, 'r', encoding=encoding)
    content = file.read()
    return content

# 写入文件内容。覆盖
def writeFile(filename, content, encoding='utf-8'):
    file_object = open(filename, 'w', encoding=encoding)
    file_object.write(content)
    file_object.close()

# 追加文件内容
def appendContent(filename, content, encoding='utf-8'):
    file_object = open(filename, 'a', encoding=encoding)
    file_object.write(content)
    file_object.close()

# 追加文件内容，文件不存在就创建
def appendFile(filename, content, encoding='utf-8'):
    file_object = open(filename, 'a+', encoding=encoding)
    file_object.write(content)
    file_object.close()

# 列表内容写入文件。覆盖
def writeList(filename, list, toString=None, encoding='utf-8'):
    content = getListContent(list, toString)
    writeFile(filename, content, encoding=encoding)

# 列表内容转换为字符串
def getListContent(list, toString=None):
    content = ""
    for item in list:
        content = content + (toString(item) if toString != None else str(item)) + "\n"
    return content
# 列表内容写入文件。 追加 
def appendList(filename, list, toString=None, encoding='utf-8'):
    content = getListContent(list, toString)
    appendFile(filename, content, encoding=encoding)

File: templates/java/FileUtil.java

import java.util.*;
import java.io.*;

public class FileUtil {
    
    private static final String DEFAULT_CHARSET = "utf-8";
    /**
     * 文本内容写入文件，默认使用 utf-8 编码
     * @param filename
     * @param content
     */
    public static void writeFile(String filename, String content) {
        writeFile(filename, content, DEFAULT_CHARSET);
    }
    /**
     * 文本内容写入文件
     * @param filename
     * @param content
     * @param charset
     */
    public static void writeFile(String filename, String content, String charset) {
        File file = new File(filename);
        try(OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(file), charset);
        BufferedWriter bw = new BufferedWriter(osw);

        ) {
            bw.write(content);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    /**
     * 文本内容追加到文件后面，默认使用 utf-8 编码
     * @param filename
     * @param content
     */
    public static void appendFile(String filename, String content) {
        appendFile(filename, content, DEFAULT_CHARSET);
    }
    /**
     * 文本内容追加到文件后面
     * @param filename
     * @param content
     * @param charset
     */
    public static void appendFile(String filename, String content, String charset) {
        File file = new File(filename);
        try(OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(file), charset);
        BufferedWriter bw = new BufferedWriter(osw);) {
            bw.append(content);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    /**
     * 读取文件行的内容到一个列表中
     * @param filename
     * @return
     */
    public static List<String> readLines(String filename) {
        return readLines(filename, DEFAULT_CHARSET);
    }
    public static List<String> readLines(String filename, String charset) {
        File file = new File(filename);
        ArrayList<String> result = new ArrayList<>();
        if (!file.exists()) {
            return result;
        }
        try(InputStreamReader fr = new InputStreamReader(new FileInputStream(file), charset);) {
            
            BufferedReader br = new BufferedReader(fr);
            String line = br.readLine();
            while(line != null){
                result.add(line);
                line = br.readLine();
            }
            
        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }

    /**
     * 读取文件内容到一个字符串
     * @param filename
     * @return
     */
    public static String readFile(String filename) {
        return readFile(filename, DEFAULT_CHARSET);
    }
    public static String readFile(String filename, String charset) {
        File file = new File(filename);
        String result = "";
        if (!file.exists()) {
            return result;
        }
        StringBuilder sb = new StringBuilder();
        try(InputStreamReader fr = new InputStreamReader(new FileInputStream(file), charset)) {
            int readed = fr.read();
            while(readed != -1){

                sb.append((char)readed);
                readed = fr.read();
            }
            result = sb.toString();
            
        } catch (IOException e) {
            e.printStackTrace();
        }

        return result;
    }
}